examples/siro_sandbox/serialize_utils.py

```diff
@@ -26,8 +28,38 @@ def pickle_vector3(obj):
     return unpickle_vector3, (pickled_data,)
 
 
+def unpickle_vector2i(data):
+    return mn.Vector2i(data)
+
+
+def pickle_vector2i(obj):
+    pickled_data = list(obj)
+    return unpickle_vector2i, (pickled_data,)
+
+
+def unpickle_matrix4(data):
+    return mn.Matrix4(data)
+
+
+def pickle_matrix4(obj):
+    pickled_data = [list(column_tuple) for column_tuple in obj]
+    return unpickle_matrix4, (pickled_data,)
+
+
+def unpickle_ray(data):
+    return habitat_sim.geo.Ray(data[0], data[1])
+
+
+def pickle_ray(obj):
+    pickled_data = [list(obj.origin), list(obj.direction)]
+    return unpickle_ray, (pickled_data,)
+
+
 # fix for unpickable type; run once at startup
 copyreg.pickle(mn.Vector3, pickle_vector3)
+copyreg.pickle(mn.Vector2i, pickle_vector2i)
```

Comment:
Not that the serialization introduced in https://github.com/mosra/magnum-bindings/commit/f561337caa9094c4a99411fac402342343f4ecb4 would produce anything significantly better -- it still has to encode the actual type as a string for each such instance. But as long as the serialized data is an arbitrary object tree / a JSON-like structure instead of a small set of typed (multi-dimensional) arrays with many elements, there's no way around that.